File: sdk/src/TrancheFairValue.ts
import { SlotTracking } from "./SlotTracking";

export class TrancheFairValue {
    value: number[];
    slotTracking: SlotTracking;

    constructor(value: number[], slotTracking: SlotTracking) {
        this.value = value;
        this.slotTracking = slotTracking;
    }
}

File: programs/vyper-core/src/state/tranche_fair_value.rs
use super::SlotTracking;
use anchor_lang::prelude::*;

#[repr(C, align(8))]
#[derive(AnchorDeserialize, AnchorSerialize, Clone, Copy, Debug, Default)]
pub struct TrancheFairValue {
    /// tranches [senior, junior] fair values expressed in bps
    pub value: [[u8; 16]; 2],
    pub slot_tracking: SlotTracking,
}

impl TrancheFairValue {
    pub const LEN: usize = 16*2 + // pub value: [[u8; 16];2],
    SlotTracking::LEN; // pub slot_tracking: SlotTracking
}
